src/AtelierTomato.Markov.Service.Discord/DiscordSentenceBuilder.cs

```diff
@@ -7,9 +7,32 @@ namespace AtelierTomato.Markov.Service.Discord
 {
 	public class DiscordSentenceBuilder
 	{
-		public static async Task<IEnumerable<Sentence>> Build(IGuild? guild, IChannel channel, ulong messageID, ulong userID, DateTimeOffset date, IEnumerable<string> sentenceTexts, string instance = "discord.com")
+		private readonly DiscordObjectOIDBuilder discordObjectOIDBuilder;
+		public DiscordSentenceBuilder(DiscordObjectOIDBuilder discordObjectOIDBuilder)
 		{
-			DiscordObjectOID OID = await DiscordObjectOIDBuilder.Build(guild, channel, messageID, instance);
+			this.discordObjectOIDBuilder = discordObjectOIDBuilder;
+		}
+
+		public async Task<IEnumerable<Sentence>> Build(IGuild guild, IGuildChannel channel, ulong messageID, ulong userID, DateTimeOffset date, IEnumerable<string> sentenceTexts, string instance = "discord.com")
+		{
+			DiscordObjectOID OID = (await discordObjectOIDBuilder.Build(guild, channel, instance)).WithMessage(messageID);
+			AuthorOID author = new(ServiceType.Discord, instance, userID.ToString(CultureInfo.InvariantCulture));
+			return sentenceTexts.Select((text, index) =>
+				new Sentence(
+					OID.WithSentence(index),
+					author,
+					date,
+					text
+				)
+			);
+		}
+		public static IEnumerable<Sentence> Build(IChannel channel, ulong messageID, ulong userID, DateTimeOffset date, IEnumerable<string> sentenceTexts, string instance = "discord.com")
+		{
+			if (channel is IGuildChannel)
+			{
+				throw new ArgumentException($"An {nameof(IGuildChannel)} was passed while this function is only used for non-guild channels. Please use the correct overload.", nameof(channel));

```

Comment:
there is no semantic field to replace, and yet you are passing a value for one, i am shocked this doesn't throw a warning.